src/mds/agg_mds/adapters.py

```diff
@@ -66,6 +66,42 @@ def aggregate_pdc_file_count(record: list):
     return file_count
 
 
+def strip_leading_double_underscore(d):
+    """
+    Rename dictionary keys that start with '__' to start with '_' instead.
+    Can handle a single dictionary or a list of dictionaries.
+
+    Args:
+        d (dict or list): A dictionary or list of dictionaries to modify
+
+    Returns:
+        dict or list: A new dictionary/list with renamed keys
+    """
+    if isinstance(d, list):
+        return [
+            rename_single_dict(item) if isinstance(item, dict) else item for item in d
+        ]
+    elif isinstance(d, dict):
+        return rename_single_dict(d)
+    else:
+        return d
+
+
+def rename_single_dict(d):
+    """Helper function to rename keys in a single dictionary."""
+    result = {}
+    for key, value in d.items():
+        # ensure keys is a string and is only leading with __
+        if isinstance(key, str) and len(key) > 2 and key[:2] == "__" and key[2] != "_":
```

Comment:
Consider using regex match testing instead of this if logic chains, can just check if `key` is string then call re.match() on `key`
I think something like this `^__[^_][\S]+$` should work, but I haven't tested